Makefile

```diff
@@ -23,43 +23,72 @@ MANDIR        ?= $(PREFIX)/share/man
 DOCDIR        ?= $(PREFIX)/share/doc
 SYSCONFDIR    ?= /etc
 CRONDIR       ?= $(SYSCONFDIR)/cron.d
+SYSTEMDDIR    ?= /lib/systemd/system
 LOGDIR        ?= /var/log
 
+ifneq ("$(wildcard /bin/systemctl)","")
+	type=systemd
+else
+	type=cron
+endif
+
 all:
 	@echo "Nothing to compile, please choose a specific target:"
-	@echo "    install (includes install-base, install-man, and install-doc)"
+	@echo "    install (includes install-base, install-man, install-doc and install-(cron/systemd)*)"
 	@echo "    install-base"
+	@echo "    install-cron"
+	@echo "    install-systemd"
 	@echo "    install-man"
 	@echo "    install-doc"
-	@echo "    uninstall (includes uninstall-base, uninstall-man, and uninstall-doc)"
+	@echo "    uninstall (includes uninstall-base, uninstall-man, uninstall-doc and uninstall-(cron/systemd)*)"
 	@echo "    uninstall-base"
+	@echo "    uninstall-cron"
+	@echo "    uninstall-systemd"
 	@echo "    uninstall-man"
 	@echo "    uninstall-doc"
+	@echo ""
+	@echo "*If systemd is available it will (un)install systemd units instead of cron"
+	@echo "Current type detected is: $(type)"
+
+recap.cron:
+	@sed -e 's|@BINDIR@|$(BINDIR)|' src/utils/recap.cron.in > src/utils/recap.cron
 
-src/recap.cron:
-	@sed -e 's|@BINDIR@|$(BINDIR)|' src/recap.cron.in > src/recap.cron
+recap.systemd:
+	for service_file in $$( ls src/utils/*.service.in ); do \
+    sed -e 's|@BINDIR@|$(BINDIR)|' $${service_file} \
+      > $$( echo $${service_file} | sed "s,.in,,"); \
+	done
 
 clean:
-	@rm -f src/recap.cron
+	@rm -f src/util/recap.cron
+	@rm -f src/util/*.service
 
-install: install-base install-man install-doc
+install: install-base install-man install-doc install-$(type)
 
-uninstall: uninstall-base uninstall-man uninstall-doc
+uninstall: uninstall-base uninstall-man uninstall-doc uninstall-$(type)
 
-install-base: src/recap.cron
+install-base:
 	@echo "Installing scripts..."
 	@install -Dm0755 src/recap $(DESTDIR)$(BINDIR)/recap
 	@install -Dm0755 src/recaplog $(DESTDIR)$(BINDIR)/recaplog
 	@install -Dm0755 src/recaptool $(DESTDIR)$(BINDIR)/recaptool
 	@echo "Installing configuration..."
 	@install -Dm0644 src/recap.conf $(DESTDIR)$(SYSCONFDIR)/recap
-	@echo "Installing cron job..."
-	@install -Dm0644 src/recap.cron $(DESTDIR)$(CRONDIR)/recap
 	@echo "Creating log directories..."
 	@install -dm0750 $(DESTDIR)$(LOGDIR)/recap
 	@install -dm0755 $(DESTDIR)$(LOGDIR)/recap/backups
 	@install -dm0755 $(DESTDIR)$(LOGDIR)/recap/snapshots
 
+install-cron: recap.cron
+	@echo "Installing cron job..."
+	@install -Dm0644 src/utils/recap.cron $(DESTDIR)$(CRONDIR)/recap
+
+install-systemd: recap.systemd
+	@echo "Installing systemd timers and services..."
+	@install -Dm0644 src/utils/*.service $(DESTDIR)$(SYSTEMDDIR)/
+	@install -Dm0644 src/utils/*.timer $(DESTDIR)$(SYSTEMDDIR)/
```

Comment:
the ```install``` will fail to work here if the directory doesn't exist; the -D option means the trailing component has to be named explicitly (by name) as it's being done in install-cron. Here, we'll need to add a line above it:
```
@install -dm0755 $(DESTDIR)$(SYSTEMDDIR)/
@install -Dm0644 src/utils/*.service $(DESTDIR)$(SYSTEMDDIR)/
@install -Dm0644 src/utils/*.timer $(DESTDIR)$(SYSTEMDDIR)/
```

Reply:
Good catch, I've fixed this in the latest commit

Reply:
That still won't work.  The `-D` flag requires the full path to the destination file (including the file name), and creates all leading directory components.  Creating the directory first with `install -d` doesn't change that.  You either need to:

Reply:
Hmm, I may be looking at this wrong, but I see that is copying the files as expected for me:
* installing:
```bash
$ make DESTDIR=test install
Installing scripts...
Installing configuration...
Creating log directories...
Installing man pages...
Installing docs...
for service_file in $( ls src/utils/*.service.in ); do \
    sed -e 's|@BINDIR@|/usr/local/sbin|' ${service_file} \
      > $( echo ${service_file} | sed "s,.in,,"); \
done
Installing systemd timers and services...
Is recommended to enable timers and reload systemd daemon.
```

Then the content inside test looks like this:
```bash
$ find test/
test/
test/var
test/var/log
test/var/log/recap
test/var/log/recap/backups
test/var/log/recap/snapshots
test/etc
test/etc/recap
test/usr
test/usr/local
test/usr/local/share
test/usr/local/share/man
test/usr/local/share/man/man5
test/usr/local/share/man/man5/recap.5
test/usr/local/share/man/man8
test/usr/local/share/man/man8/recap.8
test/usr/local/share/man/man8/recaplog.8
test/usr/local/share/man/man8/recaptool.8
test/usr/local/share/doc
test/usr/local/share/doc/recap
test/usr/local/share/doc/recap/CHANGELOG.md
test/usr/local/share/doc/recap/README.md
test/usr/local/share/doc/recap/COPYING
test/usr/local/sbin
test/usr/local/sbin/recap
test/usr/local/sbin/recaplog
test/usr/local/sbin/recaptool
test/usr/lib
test/usr/lib/systemd
test/usr/lib/systemd/system
test/usr/lib/systemd/system/recap-onboot.timer
test/usr/lib/systemd/system/recaplog.timer
test/usr/lib/systemd/system/recaplog.service
test/usr/lib/systemd/system/recap.timer
test/usr/lib/systemd/system/recap.service
test/usr/lib/systemd/system/recap-onboot.service
```